lib/kafka/avro_producer.rb

```diff
@@ -2,13 +2,17 @@
 
 require 'avro'
 require 'kafka/producer_manager'
+require 'kafka/schema_registry/service'
+require 'logger'
 
 module Kafka
   class AvroProducer
-    attr_reader :producer
+    attr_reader :producer, :registry
 
     def initialize(producer: nil)
       @producer = producer || Kafka::ProducerManager.instance.producer
+      @registry = SchemaRegistry::Service.new
```

Comment:
I tried to stick with the service/configuration setup that other libs use to stick with the vets-api precedent but now is deviating from the singleton setup of the producer manager. There's a lot of http faraday configurations that I think are important to keep consistent throughout vets-api so I'm hesitant to break away from that. We have that ticket to evaluate the use of the singleton on the producerManager, we can use this PR as a place to discuss it or we can kick the can down to whoever picks up that ticket.

Reply:
Yes, so the tricky part about the `Kafka::ProducerManager` is that we really want to use it _like_ a Singleton in the sense we want to avoid creating multiple instances of the object. However, we want to try to remove the Ruby Singleton pattern and replacing it with something else because singletons tend to be a bit hard to work with. I think we can punt it for now and go into it when we start work on the next ticket.